### ISSUE TITLE
Cannot play AAC encoded streams

### ISSUE THREAD

**Issue by nunojsa** [kodi-pvr/pvr.iptvsimple#652]
Hi,

After the update to Matrix, the addon stopped working on AAC encoded streams. This only occurs on ARM platforms and affects multiple platforms as libreelec and OSMC. I've been discussing in OSMC support about the problem for the past months (probably year already) without any luck so I'm trying my luck directly in here... Some points on the issue:

1. Only AAC streams seem to be affected. All the others work;
2. Timeshift + catchup and the video just does not work (audio out of sync, video lags a lot…) - show duration in the OSD screen is completely wrong;
3. Timeshift + catchup disabled and audio is out of sync with video;
4. As stated, it seems to affect ARM 32bits. It works on x86 and on the 4k firestick (which I actually think is running on 32 bit mode but it is android);

I've already also tried to set "Use inputstream.adaptive for m3u8 (HLS) streams" on with no luck. To me this seems to be somehow related with [inputstream.ffmpegdirect](https://github.com/xbmc/inputstream.ffmpegdirect) as I think it's one of the biggest changes (if not the biggest) from Leia to Matrix...

Any idea about the possible issue with this?

some logs in here:

[kodi.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/9811185/kodi.log)

**Comment by phunkyfish** [comment]
if you put the url in a `.strm` file and open it from a video share does it work correctly?

**Comment by nunojsa** [comment]
I will give it a try later today... What I can say that I already tried was to use tvheadend to record a faulty channel and then play it back on my arm box. In this case, it works correctly. Maybe this already replies to what you want to know.

Anyways, I'll try the `.strm` file

**Comment by nunojsa** [comment]
> if you put the url in a .strm file and open it from a video share does it work correctly?

yes, it works fine in this way...

**Comment by phunkyfish** [comment]
Ok, for the next test can you try adding this before one of the streams in your M3U file:

`#KODIPROP:inputstream=inputstream.ffmpeg`

This will force iptvsimple to use kodi’s internal ffmpeg inputstream.

**Comment by nunojsa** [comment]
> #KODIPROP:inputstream=inputstream.ffmpeg

Also works correctly with this. It's just that catchup/timeshift is not available but I guess that is expected...

**Comment by nunojsa** [comment]
To add a bit more on the above (I'm now using the "Inputstream name" setting as a workaround):

>  It's just that catchup/timeshift is not available

Well, catchup is available but is not too useful since I cannot fast forward and go back and jumping to the EPG programs is not really reliable (maybe because of my IPTV supplier)

The other thing (and this one is the more worrying one) is that the image always freezes for a few moments when switching to a channel (in some channels the freeze - sometimes black screen - duration is more annoying than others). I don't remember this behavior before so I might try to go back to kodi 19.3 to see if it happened in there.

Anyways, I guess the main thing is still to figure why the ffmpegdirect inputstream is not working at all...

**Comment by nunojsa** [comment]
@phunkyfish did you had anytime to look further into this one?

**Comment by phunkyfish** [comment]
I wonder if trying ffmpeg 5 might yield better results. Let me re-run those PRs and you can try them. It might take a while to get them built.

**Comment by phunkyfish** [comment]
You can get the Kodi build here: https://jenkins.kodi.tv/job/BuildMulti-PR/24351/. Just click your platform and download the APK.

ffmpegdirect on ffmpeg addon testbuilds are here: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.ffmpegdirect/detail/PR-204/9/artifacts

BTW, you need to download them in the next couple of days or the builds will be deleted. All builds will be based on Kodi Nexus.

**Comment by nunojsa** [comment]
Hi,

I see there's a kodi an arm build for linux but none for the inputstream addon... Can the android build be used? Probably not...

**Comment by phunkyfish** [comment]
We don’t do Linux test builds for addons. You would need to build it yourself.

**Comment by nunojsa** [comment]
Yeah, I see that there are no artifacts for the Linux builds... Note that I said ("As stated, it seems to affect ARM 32bits. It works on x86 and on the 4k firestick (which I actually think is running on 32 bit mode but it is android)"), so the Android build would not help much :)

Anyways, is this building directly from master? I might give it a try when I have the time...

**Comment by phunkyfish** [comment]
Yes, you can just checkout the PR itself.

Kodi: https://github.com/xbmc/xbmc/pull/21248
Ffmpegdirect: https://github.com/xbmc/inputstream.ffmpegdirect/pull/204

**Comment by nunojsa** [comment]
Alright, some more updates on this one... As I moved to use more an android based streaming device (because of this issue), I just realized this is also a problem on Android. The difference is that it only happens sometimes (most of times it actually works) and when it happens, I can (sometimes) recover by fast forwarding the channel (or just stop and start the channel again - which is not great).

Another annoying thing I just discovered (I guess my service provider made some changes on the streams) is that some streams have no sound at all and they are encoded with aac_latm. I already tested changing the inputstream to `inputstream.ffmpeg` and it works fine in that case...

With this, maybe it's worth it for me to try an android build? Can you share again one with ffmpeg5?

**Comment by phunkyfish** [comment]
Ffmpegdirect: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.ffmpegdirect/detail/PR-204/11/artifacts

kodi: https://jenkins.kodi.tv/job/BuildMulti-PR/24405/

please download them soon as the builds will expire after a few days.

**Comment by nunojsa** [comment]
Arghhh... It seems I missed the builds, sorry. I did not really looked at my emails this weekend :(. I was still able to download the addon zip file but not the main .apk

**Comment by phunkyfish** [comment]
No worries. Here are some freak kodi builds: https://jenkins.kodi.tv/job/BuildMulti-PR/24434/

**Comment by nunojsa** [comment]
Alright, thanks... Already downloaded the apk. I'll try to test things soon enough
